File: 2_Cinema-RESTful/src/main/java/fr/polytech/cinemaRESTful/controllers/forms/MovieForm.java
package fr.polytech.cinemaRESTful.controllers.forms;

import java.text.DateFormat;
import java.text.ParseException;
import java.text.SimpleDateFormat;
import java.util.Date;
import java.util.Locale;

public class MovieForm {

    public static final DateFormat DATE_FORMATTER = new SimpleDateFormat("dd-MM-yyyy", Locale.getDefault());

    private int id;

    private String title;

    private int duration;

    private String releaseDate;

    private int budget;

    private int amountOfRevenue;

    private int directorId;

    private int categoryId;

    public int getId() {
        return id;
    }

    public void setId(final int id) {
        this.id = id;
    }

    public String getTitle() {
        return title;
    }

    public void setTitle(final String title) {
        this.title = title;
    }

    public int getDuration() {
        return duration;
    }

    public void setDuration(final int duration) {
        this.duration = duration;
    }

    public Date getReleaseDate() {
        if (this.releaseDate != null) {
            try {
                return DATE_FORMATTER.parse(this.releaseDate);
            } catch (ParseException e) {
            }
        }

        return null;
    }

    public void setReleaseDate(final String releaseDate) {
        this.releaseDate = releaseDate;
    }

    public int getBudget() {
        return budget;
    }

    public void setBudget(final int budget) {
        this.budget = budget;
    }

    public int getAmountOfRevenue() {
        return amountOfRevenue;
    }

    public void setAmountOfRevenue(final int amountOfRevenue) {
        this.amountOfRevenue = amountOfRevenue;
    }

    public int getDirectorId() {
        return directorId;
    }

    public void setDirectorId(final int directorId) {
        this.directorId = directorId;
    }

    public int getCategoryId() {
        return categoryId;
    }

    public void setCategoryId(final int categoryId) {
        this.categoryId = categoryId;
    }
}

File: 4_Cinema-JSP/Polytech-Cinema-Web/src/main/webapp/resources/js/movie-form.js
function checkInputs() {
	var error = false;
	
    if (document.movieForm.title.value == "") {
	    	alert("Please enter your title");
	    	error = true;
    }
    
    if (document.movieForm.duration.value == "") {
	    	alert("Please enter your duration");
	    	error = true;
    }
    
    if (document.movieForm.releaseDate.value == "") {
	    	alert("Please enter your release date");
	    	error = true;
	}
    
    if (document.movieForm.budget.value == "") {
	    	alert("Please enter your budget");
	    	error = true;
	}
    
    if (document.movieForm.amountOfRevenue.value == "") {
	    	alert("Please enter your amount of revenue");
	    	error = true;
	}
    
    if (document.movieForm.directorId.value == "") {
	    	alert("Please enter your director");
	    	error = true;
	}
    
    if (document.movieForm.categoryId.value == "") {
	    	alert("Please enter your category");
	    	error = true;
	}
    
    return !error;
}